### PR TITLE
Environment variable documentation

### PR DESCRIPTION
**Pull request by CharlieL7**
Adds env variable documentation.
Depends on #2416.

### CODE REVIEW

File: docs/dev/env_vars.rst [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+**MIGRAPHX_TRACE_MATCHES**
+
+Set to "1" to print the matcher that matches an instruction and the matched instruction.
+Set to "2" and use the ``MIGRAPHX_TRACE_MATHCES_FOR`` flag to filter out results.
+
+**MIGRAPHX_TRACE_MATCHES_FOR**
+
+Set to a string of what you want to filter matched results of.
+TODO: need to figure out what this can do
```
Comment (umangyadav) [inline docs/dev/env_vars.rst:31]: It can be set to name of any matcher and only traces for that matcher will be printed out.

**Comment by umangyadav** [comment]
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/commit/3c160a3fc0ff7d3b243faf480bc2c89f6ca12e60 
One more ENV var to add

**Review by TedThemistokleous — APPROVED** [review]

File: docs/dev/env_vars.rst [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+**MIGRAPHX_DEBUG_SAVE_TEMP_DIR**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Make it so the created temporary directories are not deleted.
+
+**MIGRAPHX_GPU_DEBUG**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Adds the option ``-DMIGRAPHX_DEBUG`` when compiling HIPRTC.
```
Comment (umangyadav) [inline docs/dev/env_vars.rst:175]:
```suggestion
Internally, this adds the option ``-DMIGRAPHX_DEBUG`` when compiling GPU kernels.
```

File: docs/dev/env_vars.rst [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Print the compile pass and the program after the pass.
+
+**MIGRAPHX_TIME_PASSES**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Time the compile passes.
+
+
+GPU Kernels JIT compilation debugging
```
Comment (umangyadav) [inline docs/dev/env_vars.rst:154]: These options are applicable for both hiprtc and hipclang.

File: docs/dev/env_vars.rst [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Enable the ``layout_nhwc`` pass.
+
+**MIGRAPHX_ENABLE_CK**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Enable using the Composable Kernels library.
+
+**MIGRAPHX_ENABLE_MLIR**
```
Comment (causten) [inline docs/dev/env_vars.rst:120]: @pfultz2  Should this exist?  It is enabled by default now.  Perhaps the comments need to mention what release it was deprecated at?

File: docs/dev/env_vars.rst [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+**MIGRAPHX_ENABLE_NHWC**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Enable the ``layout_nhwc`` pass.
+
+**MIGRAPHX_ENABLE_CK**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Enable using the Composable Kernels library.
```
Comment (causten) [inline docs/dev/env_vars.rst:118]:
```suggestion
Enable using the Composable Kernels library. Should be used in conjunction with MIGRAPHX_DISABLE_MLIR=1
```

File: docs/dev/env_vars.rst (hunk 4 above)
Comment (umangyadav) [reply docs/dev/env_vars.rst:120]: This variable doesn't exist anymore.

Comment (umangyadav) [reply docs/dev/env_vars.rst:120]: There is one more ENV var `MIGRAPHX_ENABLE_EXTRA_MLIR`

File: docs/dev/env_vars.rst [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+
+**MIGRAPHX_GPU_DUMP_SRC**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Dump the HIPRTC source files compiled.
+
+**MIGRAPHX_GPU_DUMP_ASM**
+
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Dump the HIPRTC assembly.
```
Comment (umangyadav) [inline docs/dev/env_vars.rst:190]: It doesn't print for HIPRTC, This flag is only valid for hip-clang

File: docs/dev/env_vars.rst (hunk 4 above)
Comment (causten) [reply docs/dev/env_vars.rst:120]: Set to "1", "enable", "enabled", "yes", or "true" to use.
Enables additional opportunities to use MLIR that may improve performance.

File: docs/dev/env_vars.rst [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+Set to "1", "enable", "enabled", "yes", or "true" to use.
+Use tuning for Composable Kernels.
+
+Testing 
+------------
+
+**MIGRAPHX_TRACE_TEST_COMPILE**
+
+Set to the target that you want to trace the compilation of (ex. "gpu", "cpu").
+Prints the compile trace for the given target for the verify tests.
```
Comment (umangyadav) [inline docs/dev/env_vars.rst:284]: This flag shouldn't be used in conjunction with `MIGRAPHX_TRACE_COMPILE`. For the verify tests only use `TRACE_TEST_COMPILE

**Review by umangyadav — APPROVED** [review]

**Pull request merged**